lib/shared.bash

```diff
@@ -92,7 +92,7 @@ secret_exists() {
 secret_download() {
   local server="$1"
   local key="$2"
-  if ! _secret=$(vault kv get -address="$server" -field=data -format=yaml "$key" | sed 's/: /=/g' ); then
+  if ! _secret=$(vault kv get -address="$server" -field=data -format=yaml "$key" | sed -r 's/: /=/g; s/\"/\\"/g; s/\$/\\$/g; s/=(.*)$/=\"\1\"/g' ); then
```

Comment:
I have one issue with the first replacement as it messes up values that contain `:  `. For example, a value like `VARIABLE=\"making: life: complicated\"` would get changed to `VARIABLE=\"making=life=complicated\"`

Unfortunately, I'm not too familiar with vault to offer a suggestion about this here :(

Reply:
Simple solution would be to remove the global flag in the initial regex pattern (`s/: /=/`) so that it only catches the first colon, which from a quick test appears to work as expected
```
➜  ~ echo "this: is: a: test:" | sed -r 's/: /=/; s/\"/\\"/g; s/\$/\\$/g; s/=(.*)$/=\"\1\"/g'
this="is: a: test:"
```
Also works in instances with multiple lines which I tested with an actual `vault kv get -format=yaml -field=data` command

Reply:
Nice catch on the regex @fd-jonathanlinn - updated the pattern and the test case - looks good!